### ISSUE TITLE
Consider versioning pinning support?

### ISSUE THREAD

**Issue by lread** [liquidz/antq#78]
My scenario:

The minimum version of Clojure for rewrite-clj is 1.9.0.
This is therefore what ends up in my pom.xml.
My deps.edn, though, references other versions of Clojure to support testing.

I like that antq tells me when there is a newer version of Clojure when it analyzes my deps.edn.
But when it analyzes my pom.xml, I don't need to know that there is a newer version of Clojure.

I started off by ignoring this report, but then decided to add some [custom version pinning support via my babashka script](https://github.com/clj-commons/rewrite-clj/blob/6c53f795ceebb8349b8a4d432ba9bdc557d58ccf/script/outdated.clj#L13-L26).

That's one approach, but I guess I'd probably also be happy if I knew that deps in my deps.edn are not outdated (we already have this) and that deps in my pom.xml are in synch with my deps.edn.  But maybe such a check is out of the realm of antq.

I'm quite OK with my little babashka script hack but I think that this this scenario might be common and that's why I've raised it for your consideration. I expect that most folks would end up just excluding their pom.xml from antq analysis.

So to recap ideas for this scenario:
1. No antq change: Antq analyzes pom.xml, user ignores report about outdated Clojure.
2. No antq change: User excludes pom.xml from antq analysis, but might miss that their pom.xml deps are out of synch with their deps.edn.
3. Antq adds version pinning support. For this scenario, it would have to be scoped to pom.xml.
4. Antq adds synch check support. Is my pom.xml out of date against my deps.edn?
5. No antq change: Lee carries on happily with his custom pinning support in his babashka script.
6. Some other idea

**Comment by liquidz** [comment]
@lread Thanks for your suggestion!

Since it is a case testing multiple versions and talking about the minimum versions of them, I'm assuming this is a problem limited to Clojure itself. (right?)

> But maybe such a check is out of the realm of antq.

I think so too for now.
In my opinion, it is a problem about dependencies between files. (in this case, deps.edn and pom.xml)

> 4. Antq adds synch check support. Is my pom.xml out of date against my deps.edn?

If I had to do it, it would be this.
But, for me, it should be solved by specific tools such as `Make` or `Babashka' task runner` (e.g. `fs/modified-since`).

I think it would be better that antq excludes pom.xml and we sync pom.xml with deps.edn in our development flow as follows.

```makefile
pom.xml: deps.edn
	clojure -Spom

outdated: pom.xml
	clojure -M:outdated --skip=pom
```

BTW if we assume that we have a version above the minimum version in deps.edn, we can use `--exclude=org.clojure/clojure@1.9.0` as workaround.

**Comment by lread** [comment]
Thanks for the reply @liquidz! 

> Since it is a case testing multiple versions and talking about the minimum versions of them, I'm assuming this is a problem limited to Clojure itself. (right?)

Yes, good point! 

> BTW if we assume that we have a version above the minimum version in deps.edn, we can use `--exclude=org.clojure/clojure@1.9.0` as workaround.

Hmm... that is an interesting idea!

Much thanks for reviewing my scenario, and am happy with the way antq behaves now.

**Comment by lread** [comment]
@liquidz I am uncertain of proper etiquette, but I think I should close this issue?  
If this was not the case, apologies, please reopen.

**Comment by liquidz** [comment]
@lread I wasn't sure if the problem had been solved, so I left it.
It's OK to close. Thanks!
